### PR TITLE
Introduce setting `create_table_empty_primary_key_by_default`

### LINKED ISSUE #55023: Default ORDER BY key
I found the setting to enable default table engine, but you can only set default engine = MergeTree

therefore, one must always write CREATE TABLE db.table **ORDER BY tuple() a**s  select .. unless one uses the log engine.

I would hope for an option MergeTree ORDER BY tuple() as a default engine, because then the create table statement would look like one in any other SQL language.

### PR DESCRIPTION
**Pull request by srikanthccv**
Fixes #55023 

### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Introduce setting `create_table_empty_primary_key_by_default` for default `ORDER BY ()`

### Documentation entry for user-facing changes

If set to true and the table of MergeTree family does not contain ORDER BY and PRIMARY KEY, then an empty tuple is used by default

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 0f2e0ade996a9c723b13676d44d168e9a955b3fd with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/55899/0f2e0ade996a9c723b13676d44d168e9a955b3fd/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6598429823">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/55899/0f2e0ade996a9c723b13676d44d168e9a955b3fd/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Comment by alexey-milovidov** [comment]
Need to add tests.

**Comment by alexey-milovidov** [comment]
Tests should include:
- a check that it throws an exception when there is no primary key, and the setting is not enabled;
- a check that it successfully creates a table when there is no primary key, and the setting is enabled;
- a check that it does not get confused when you specify a primary on a per-column basis (x String PRIMARY KEY, y UInt64 PRIMARY KEY) and the setting is enabled;

**Comment by alexey-milovidov** [comment]
Functional tests should be sufficient; no need for integration tests. See `tests/queries/0_stateless`.

**Comment by srikanthccv** [comment]
I was unsure about that but yes, makes sense. Let me update.

File: docs/en/engines/table-engines/mergetree-family/mergetree.md [hunk 1]
```diff
@@ -73,7 +73,7 @@ A tuple of column names or arbitrary expressions. Example: `ORDER BY (CounterID,
 
 ClickHouse uses the sorting key as a primary key if the primary key is not defined explicitly by the `PRIMARY KEY` clause.
 
-Use the `ORDER BY tuple()` syntax, if you do not need sorting. See [Selecting the Primary Key](#selecting-the-primary-key).
+Use the `ORDER BY tuple()` syntax, if you do not need sorting Or set `create_table_empty_primary_key_by_default` to `true` to use the `ORDER BY tuple()` syntax by default. See [Selecting the Primary Key](#selecting-the-primary-key).
```
Comment (Avogar) [inline docs/en/engines/table-engines/mergetree-family/mergetree.md:76]:
```suggestion
Use the `ORDER BY tuple()` syntax, if you do not need sorting, or set `create_table_empty_primary_key_by_default` to `true` to use the `ORDER BY tuple()` syntax by default. See [Selecting the Primary Key](#selecting-the-primary-key).
```

**Review by Avogar — APPROVED** [review]

**Comment by Avogar** [comment]
Seems like some problem with CI, please, update your branch with master. We cannot merge PR with failed build checks

**Pull request merged**